options/components/webBottomContent/WebBottomContent.jsx

```diff
@@ -4,31 +4,64 @@ import { WebSearchContext } from "../../context/WebSearchContext";
 
 export default function WebBottomContent() {
   return (
-    <div className="w-full h-[calc(100vh-200px)] overflow-hidden max-w-5xl mx-auto my-0">
+    <div className="w-full h-[calc(100vh-200px)] overflow-hidden lg:max-w-5xl lg:px-0 md:max-w-screen-md sm:max-w-screen-sm px-2 mx-auto my-0">
       <WebUrlNewList />
     </div>
   );
 }
 
 function WebUrlNewList() {
-  const { bookmarkList } = useContext(WebSearchContext);
+  const { bookmarkList, hasSearchResult } = useContext(WebSearchContext);
 
+  function faviconURL(u) {
+    const url = new URL(chrome.runtime.getURL("/_favicon/"));
+    url.searchParams.set("pageUrl", u);
+    url.searchParams.set("size", "32");
+    return url.toString();
+  }
+  console.log(hasSearchResult);
   return (
-    <div className="w-full h-[calc(100vh-200px)] overflow-y-auto">
+    <div className="w-full overflow-y-auto grid lg:grid-cols-3 md:grid-cols-2 sm:grid-cols-2 gap-4">
       {bookmarkList.map((url, index) => (
         <div
-          className="w-full flex"
+          className="w-full min-h-10 flex bg-white rounded-lg p-3 hover:bg-gray-300 hover:shadow-lg duration-100"
           key={index}
         >
           <a
-            className="w-full max-w-[calc(100%-10px)] flex-grow overflow-hidden text-ellipsis whitespace-nowrap"
+            className="w-full max-w-[calc(100%-0px)] flex-grow overflow-hidden text-ellipsis whitespace-nowrap font-bold"
             href={`${url.url}`}
             target="_blank"
           >
+            <img
+              className="mr-2 inline-block w-3 h-3"
+              src={faviconURL(url.url)}
+            />
             {url.title}
+            {hasSearchResult ? <HighlightKeyword url={url} /> : ""}
           </a>
         </div>
       ))}
     </div>
   );
 }
+
+function HighlightKeyword({ url }) {
+  const { keyword } = useContext(WebSearchContext);
+  console.log(keyword);
+
+  return (
+    <div className="mt-3 pt-3 border-t font-normal w-full max-w-[calc(100%-0px)] overflow-hidden text-ellipsis whitespace-nowrap">
+      {url.urlText.includes(keyword) ? (
+        <>
+          {url.urlText.split(keyword)[0]}
+          <span className="bg-blue-800 rounded-lg px-1 inline-block text-white mx-px">
+            {keyword}
+          </span>
+          {url.urlText.split(keyword)[1]}
+        </>
+      ) : (
+        ""
+      )}
+    </div>
+  );
```

Comment:
url.urlText에 포함된 keyword 기준으로 배열 분리 후 그 사이에 색상이 포함된 keyword를 추가하는 방법인 것 같습니다.
좋은 코드라고 생각이 되지만 단점으로는 keyword가 복수 포함되어 있는 상황을 깜빡하신것 같습니다.
그래서 저는 복수의 keyword가 포함되어 있을 때도 사용 가능한 아래 코드를 추천 드립니다.

아래의 코드는 map으로 반복하며 복수의 keyword에 대응하도록 했습니다. map안 조건문은 맨 처음 keyword가 존재 시 대응하기 위한 조건문 입니다.
```suggestion
  return (
    <div className="mt-3 pt-3 border-t font-normal w-full max-w-[calc(100%-0px)] overflow-hidden text-ellipsis whitespace-nowrap">
      {url.urlText.includes(keyword) ? (
        url.urlText.split(keyword).map((item, index) => {
          if (index === 0 && !item) {
            return null;
          } else if (index === 0 && item) {
            return <span>{item}</span>
          } else {
            return (
              <span>
                <span className="bg-blue-800 rounded-lg px-1 inline-block text-white mx-px">
                {keyword} 
                </span>
                {item}
              </span>
              )
            }
          }
        })
      }
    </div>
  );
```